### PR TITLE
[16.0][MIG] project_task_stage_state:migration module

### PR DESCRIPTION
**Pull request by valjaro**
Migration module project_task_stage_state from v15 to v16

### CODE REVIEW

**Review by chusamo — APPROVED** [review]
LGTM

**Review by zamberjo — APPROVED** [review]
technical review

**Review by AungKoKoLin1997 — COMMENTED** [review]
Can you please update USAGE.rst and I would like you to follow this https://github.com/OCA/project/pull/1009#issuecomment-1305647189.

File: project_task_stage_state/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2014 Daniel Reis
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "Add State field to Project Stages",
+    "version": "16.0.1.0.1",
```
Comment (dalonsod) [inline project_task_stage_state/__manifest__.py]: Wrong version:
```suggestion
    "version": "16.0.1.0.0",
```

File: project_task_stage_state/models/project_task_type.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    ("cancelled", "Cancelled"),
+]
+
+
+class ProjectTaskType(models.Model):
+    """Added state in the Project Task Type."""
+
+    _inherit = "project.task.type"
+
+    state = fields.Selection(_TASK_STATE)
```
Comment (dalonsod) [inline project_task_stage_state/models/project_task_type.py]: Could you attend https://github.com/OCA/project/pull/1009#discussion_r1017752736 ?

**Review by dalonsod — CHANGES_REQUESTED** [review]

File: project_task_stage_state/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
+To use this module, you need to:
+
+#. Go to Project -> Configuration -> Stages and click on a stage
+#. Select the state you would like to associate that stage with from the dropdown "State" menu
+#. Save your changes
+#. Go to Project -> Dashboard and click on a project
+#. Click on task in the stage you just edited
+#. Under the "Extra Info" tab, you can see the "State" field for that task
```
Comment (dalonsod) [inline project_task_stage_state/readme/USAGE.rst]: Please update this line, because this field is now in main section of the form

**Review by dalonsod — CHANGES_REQUESTED** [review]
Tested on runboat 👍 , see comment

**Review by navarrorico — APPROVED** [review]

**Review by dalonsod — APPROVED** [review]

**Comment by leemannd** [comment]
Hello @valjaro Have you followed these steps? https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0#technical-method-to-migrate-a-module-from-150-to-160-branch
It seems strange to me that older commits are missing into this migration

**Review by AungKoKoLin1997 — COMMENTED** [review]
Why this PR includes project_parent and it's commit history?

**Comment by mathiasfrancke** [comment]
Hello @valjaro ,
may I ask why this PR was closed? Do you have any suggestions here? Can I help so we got this module migrated?

Best,
Mathias

**Comment by houzefa-abba** [comment]
Hi thanks for this migration

If someone (@valjaro / @QubiQ / @QubiQBcn?) works on fixing commit history, please also note there seems to have been a mix-up in the latest ones as [the one titled "Migration to 15.0"](https://github.com/QubiQ/project/commit/0e3da909d5c0ee479f824f1d947562b89ce70817) updates manifest to 16.0 and [the one titled "Migration to 16.0"](https://github.com/QubiQ/project/commit/96cf0214c1eae3029da11d6a5b5d3d858eb6633f) only touches `setup/_metapackage/VERSION.txt`

**Pull request closed without merging**